specs/RuntimeSpec/Actions.md

```diff
@@ -32,35 +43,48 @@ _Requirements:_
 - sets Account `storage_usage` to `account_cost` (genesis config)
 - sets Account `storage_paid_at` to the current block height
 
-NOTE: for the all subsequent actions in the transaction the `signer_id` becomes `receiver_id` until [DeleteAccountAction](#DeleteAccountAction). It allows to execute actions on behalf of the just created account.
-
+**Errors**:
+- If the action tries to create a top level account whose length is no greater than 32 characters, and `predecessor_id` is not
+`registrar_account_id`, the following error will be returned
 ```rust
-pub struct CreateAccountAction {}
+/// A top-level account ID can only be created by registrar.
+CreateAccountOnlyByRegistrar {
+    account_id: AccountId,
+    registrar_account_id: AccountId,
+    predecessor_id: AccountId,
+}
 ```
 
-## DeployContractAction
-
-_Requirements:_
+- If the action tries to create an account that is neither a top-level account or a subaccount of `predecessor_id`,
+the following error will be returned
+```rust
+/// A newly created account must be under a namespace of the creator account
+CreateAccountNotAllowed { account_id: AccountId, predecessor_id: AccountId },
+```
 
-- _`tx.signer_id` to be equal to `receiver_id`_
-- _`tx.public_key` to be `AccessKeyPermission::FullAccess` for the `singer_id`_
+NOTE: for the all subsequent actions in the transaction the `signer_id` becomes `receiver_id` until [DeleteAccountAction](#DeleteAccountAction). It allows to execute actions on behalf of the just created account.
 
-**Outcome**:
-- sets a code for account
+## DeployContractAction
 
 ```rust
 pub struct DeployContractAction {
     pub code: Vec<u8>, // a valid WebAssembly code
 }
 ```
 
-## FunctionCallAction
-
 _Requirements:_
 
-- _`tx.public_key` to be `AccessKeyPermission::FullAccess` or `AccessKeyPermission::FunctionCall`_
+- _`tx.signer_id` to be equal to `receiver_id`_
```

Comment:
In runtime we used a concept of an actor_id. This is the temporary account ID that is either the `predecessor_id` (not `signer_id`) or the current account ID. By default it's the predecessor, but if the first action is `CreateAccount` and the account doesn't exist, it changes the actor to current account ID `receiver_id`